src/IECoreArnold/Renderer.cpp

```diff
@@ -574,26 +574,47 @@ class ArnoldOutput : public IECore::RefCounted
 			}
 			else
 			{
-				std::string arnoldType = "RGB";
+				std::string colorType = "RGB";
 				if( parameter<bool>( output->parameters(), "includeAlpha", false ) )
 				{
-					arnoldType = "RGBA";
+					colorType = "RGBA";
 				}
 
 				vector<std::string> tokens;
 				IECore::StringAlgo::tokenize( m_data, ' ', tokens );
+
 				if( tokens.size() == 2 )
 				{
 					if( tokens[0] == "color" )
 					{
-						m_data = tokens[1] + " " + arnoldType;
+						m_data = tokens[1] + " " + colorType;
 					}
 					else if( tokens[0] == "lpe" )
 					{
 						m_lpeValue = tokens[1];
-						m_data = m_lpeName + " " + arnoldType;
+						m_data = m_lpeName + " " + colorType;
+					}
+					else if( tokens[0] == "float" || tokens[0] == "int" || tokens[0] == "uint" )
+					{
+						// Cortex convention is `<type> <name>`. Arnold
+						// convention is `<name> <TYPE>`.
+						m_data = tokens[1] + " " + boost::to_upper_copy( tokens[0] );
+					}
+					else
+					{
+						IECore::msg(
+							IECore::Msg::Warning, "ArnoldRenderer",
+							boost::format( "Unknown data type \"%1%\" for output \"%2%\"" ) % tokens[0] % name
+						);
```

Comment:
So, we still support Arnold syntax, but we now print this warning?  I don't recall exactly where, but I feel like we may be using the Arnold syntax somewhere at IE ( possibly because float wasn't supported with Gaffer syntax, or maybe because just using ass syntax felt easiest ) - I can double check this tomorrow.  I feel like at least the wording here should probably change: "Unknown data type" makes me think "The data type was unrecognized so nothing happened", not "the data was passed through directly to Arnold".

Reply:
Correct. I've added some clarifying comments in https://github.com/GafferHQ/gaffer/pull/4648/commits/edb7492ee5812a4704395da0e8a34396d1f59263. Hopefully the "unknown data type" message makes sense given that additional context?